src/Thentos/Frontend/Handlers.hs

```diff
@@ -1,47 +1,56 @@
+{-# LANGUAGE BangPatterns           #-}
 {-# LANGUAGE DataKinds              #-}
-{-# LANGUAGE ScopedTypeVariables    #-}
+{-# LANGUAGE DeriveDataTypeable     #-}
+{-# LANGUAGE LambdaCase             #-}
 {-# LANGUAGE OverloadedStrings      #-}
-{-# LANGUAGE BangPatterns           #-}
+{-# LANGUAGE ScopedTypeVariables    #-}
 
 module Thentos.Frontend.Handlers where
 
 import Control.Applicative ((<$>))
 import Control.Concurrent.MVar (MVar)
-import Control.Lens ((^.))
+import Control.Lens ((^.), (.~), (%~))
 import Control.Monad.IO.Class (liftIO)
 import Control.Monad.State.Class (gets)
 import Crypto.Random (SystemRNG)
 import Data.Acid (AcidState)
-import Data.ByteString (ByteString)
+import Data.ByteString.Builder (Builder, toLazyByteString)
 import Data.Configifier ((>>.), Tagged(Tagged))
 import Data.Functor.Infix ((<$$>))
 import Data.Monoid ((<>))
 import Data.Proxy (Proxy(Proxy))
-import Data.String.Conversions (cs)
+import Data.String.Conversions (SBS, cs)
 import Data.Text.Encoding (decodeUtf8, decodeUtf8', encodeUtf8)
 import LIO.DCLabel ((/\), (\/))
-import Snap.Snaplet.Session (commitSession, setInSession, getFromSession, resetSession)
-import Snap.Snaplet.AcidState (getAcidState, update)
 import Snap.Blaze (blaze)
-import System.Log.Missing (logger)
-import Snap.Core (rqURI, getParam, getsRequest, redirect', parseUrlEncoded, printUrlEncoded, modifyResponse, setResponseStatus)
 import Snap.Core (getResponse, finishWith, urlEncode, urlDecode)
+import Snap.Core (rqURI, getParam, getsRequest, redirect', modifyResponse, setResponseStatus)
+import Snap.Snaplet.AcidState (getAcidState, update)
+import Snap.Snaplet.Session (commitSession, setInSession, getFromSession, resetSession)
 import Snap.Snaplet (with)
+import System.Log.Missing (logger)
 import System.Log (Priority(DEBUG, INFO, WARNING, CRITICAL))
 import Text.Digestive.Snap (runForm)
+import URI.ByteString (URI(..), RelativeRef(..), URIParserOptions)
+import URI.ByteString (parseURI, parseRelativeRef, laxURIParserOptions, serializeURI, serializeRelativeRef)
+import URI.ByteString (rrPathL, uriQueryL, queryPairsL)
+
+-- We are using an experimental version of URI.ByteString:
```

Comment:
yes, travis doesn't find this in the cache, but the script was supposed to fall back to installing missings.

I would prefer not to fix this in the travis cache until we have a working release on hackage.  Shouldn't be long: https://github.com/Soostone/uri-bytestring/pull/4